File: P_2/Examen_P2/Napakalaki_Java_P2/src/napakalaki/CardDealer.java
// MADE BY: JOSE MIGUEL HDEZ GARCIA & JAIRO ABRIL MOYA

package napakalaki;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

/**
 *
 * @author ixjosemi
 */

public class CardDealer {
    
    private static CardDealer instance = null;
    
    private CardDealer(){
        
    }
    
    // Declaramos el array que va a contener todos los tesoros
    static ArrayList <Treasure> unusedTreasures = new ArrayList();
    
    // Declaramos el array que va a contener los tesoros usados
    static ArrayList <Treasure> usedTreasures = new ArrayList();
    
    // Declaramos el array que va a contener todos los monstruos
    static ArrayList <Monster> unusedMonsters = new ArrayList();
    
    // Declaramos el array que va a contener los monstruos usados
    static ArrayList <Monster> usedMonsters = new ArrayList();
    
    
    // Declaramos todos los tesoros
    private void initTreasureCardDeck(){
        
       // ¡Si, mi amo!
       unusedTreasures.add(new Treasure("¡Si, mi amo!", 4, TreasureKind.HELMET));
       
       // Botas de investigacion
       unusedTreasures.add(new Treasure("Botas de investigacion", 3, TreasureKind.SHOES));
       
       // Capucha de Cthulhu
       unusedTreasures.add(new Treasure("Capucha de Cthulhu", 3, TreasureKind.HELMET));
       
       // A prueba de babas
       unusedTreasures.add(new Treasure("A prueba de babas", 2, TreasureKind.ARMOR));
       
       // Botas de lluvia acida
       unusedTreasures.add(new Treasure("Botas de lluvia acida", 1, TreasureKind.SHOES));
       
       // Casco minero
       unusedTreasures.add(new Treasure("Casco minero", 2, TreasureKind.HELMET));
       
       // Ametralladora ACME
       unusedTreasures.add(new Treasure("Ametralladora ACME", 4, TreasureKind.BOTHHANDS));
       
       // Camiseta de la ETSIIT
       unusedTreasures.add(new Treasure("Camiseta de la ETSIIT", 1, TreasureKind.ARMOR));
       
       // Clavo de rail ferroviario
       unusedTreasures.add(new Treasure("Clavo de rail ferroviario", 3, TreasureKind.ONEHAND));
       
       // Cuchillo de sushi arcano
       unusedTreasures.add(new Treasure("Cuchillo de sushi arcano", 2, TreasureKind.ONEHAND));
       
       // Fez alopodo
       unusedTreasures.add(new Treasure("Fez alopodo", 3, TreasureKind.HELMET));
       
       // Hacha prehistorica
       unusedTreasures.add(new Treasure("Hacha Prehistorica", 2, TreasureKind.ONEHAND));
       
       // El aparato del Pr. Tesla
       unusedTreasures.add(new Treasure("El aparato del Pr. Tesla", 4, TreasureKind.ARMOR));
       
       // Gaita
       unusedTreasures.add(new Treasure("Gaita", 4, TreasureKind.BOTHHANDS));
       
       // Insecticida
       unusedTreasures.add(new Treasure("Insecticida", 2, TreasureKind.ONEHAND));
       
       // Escopeta de 3 cañones
       unusedTreasures.add(new Treasure("Escopeta de 3 cañones", 4, TreasureKind.BOTHHANDS));
       
       // Garabato mistico
       unusedTreasures.add(new Treasure("Garabato mistico", 2, TreasureKind.ONEHAND));
       
       // La rebeca metalica
       unusedTreasures.add(new Treasure("La rebeca metalica", 2, TreasureKind.ARMOR));
       
       // Lanzallamas
       unusedTreasures.add(new Treasure("Lanzallamas", 4, TreasureKind.BOTHHANDS));
       
       // Necrocomicon
       unusedTreasures.add(new Treasure("Necrocomicon", 1, TreasureKind.ONEHAND));
       
       // Necronomicon
       unusedTreasures.add(new Treasure("Necronomicon", 5, TreasureKind.BOTHHANDS));
       
       // Linterna a 2 manos
       unusedTreasures.add(new Treasure("Linterna a 2 manos", 3, TreasureKind.BOTHHANDS));
       
       // Necrognomicon
       unusedTreasures.add(new Treasure("Necrognomicon", 2, TreasureKind.ONEHAND));
       
       // Necrotelecom
       unusedTreasures.add(new Treasure("Necrotelecom", 2, TreasureKind.HELMET));
       
       // Mazo de los antiguos
       unusedTreasures.add(new Treasure("Mazo de los antiguos", 3, TreasureKind.ONEHAND));
       
       // Necroplayboycon
       unusedTreasures.add(new Treasure("Necroplayboycon", 3, TreasureKind.ONEHAND));
       
       // Porra preternatural
       unusedTreasures.add(new Treasure("Porra preternatural", 2, TreasureKind.ONEHAND));
       
       // Shogulador
       unusedTreasures.add(new Treasure("Shogulador", 1, TreasureKind.BOTHHANDS));
       
       // Varita de atizamiento
       unusedTreasures.add(new Treasure("Varita de atizamiento", 3, TreasureKind.ONEHAND));
       
       // Tentaculo de pega
       unusedTreasures.add(new Treasure("Tentaculo de pega", 2, TreasureKind.HELMET));
       
       // Zapato deja-amigos
       unusedTreasures.add(new Treasure("Zapato deja-amigos", 1, TreasureKind.SHOES));
       
    }
    
    // Declaramos todos los monstruos
    private void initMonsterCardDeck(){
        
         // 3 Byakhee de bonanza
        BadConsequence badConsequence1 = new BadConsequence ("Pierdes tu armadura "
            + "visible y otra oculta ", 0,
            new ArrayList (Arrays.asList(TreasureKind.ARMOR)),
            new ArrayList (Arrays.asList(TreasureKind.ARMOR)));
        Prize prize1 = new Prize (4, 2);
        unusedMonsters.add(new Monster("3 Byakhee de bonanza", 8, badConsequence1, prize1));

        // Tenochtitlan
        BadConsequence badConsequence2 = new BadConsequence ("Embobados con el "
            + "lindo primigenio te descartas de tu casco visible", 0,
            new ArrayList (Arrays.asList(TreasureKind.HELMET)),
            new ArrayList (Arrays.asList()));
        Prize prize2 = new Prize (1, 1);
        unusedMonsters.add(new Monster("Tenochtitlan", 2, badConsequence2, prize2));

        // El sopor de Dunwich
        BadConsequence badConsequence3 = new BadConsequence ("El primordial bostezo "
            + "contagioso. Pierdes el calzado visible", 0,
            new ArrayList (Arrays.asList(TreasureKind.SHOES)),
            new ArrayList (Arrays.asList()));
        Prize prize3 = new Prize (1, 1);
        unusedMonsters.add(new Monster("El sopor de Dunwich", 2, badConsequence3, prize3));

        // Demonios de Magaluf
        BadConsequence badConsequence4 = new BadConsequence ("Te atrapan para "
            + "llevarte de fiesta y te dejan caer en mitad del vuelo. "
            + "Descarta 1 mano visible y 1 mano oculta", 0,
            new ArrayList(Arrays.asList(TreasureKind.ONEHAND)),
            new ArrayList(Arrays.asList(TreasureKind.ONEHAND)));
        Prize prize4 = new Prize (4, 1);
        unusedMonsters.add (new Monster("Demonios de Magaluf", 2, badConsequence4, prize4));

        // El gorrón en el umbral
        BadConsequence badConsequence5 = new BadConsequence ("Pierdes todos tus "
            + "objetos visibles", 0, BadConsequence.MAXTREASURES, 0);
        Prize prize5 = new Prize (3, 1);
        unusedMonsters.add (new Monster("El gorrón en el umbral", 13, badConsequence5, prize5));

        // H.P. Munchcraft
        BadConsequence badConsequence6 = new BadConsequence ("Pierdes la armadura "
            + "visible", 0,
            new ArrayList (Arrays.asList(TreasureKind.ARMOR)),
            new ArrayList (Arrays.asList()));
        Prize prize6 = new Prize (2, 1);
        unusedMonsters.add (new Monster("H.P. Munchcraft", 6, badConsequence6, prize6));

        // Necrófago
        BadConsequence badConsequence7 = new BadConsequence ("Sientes bichos bajo "
            + "la ropa. Descarta la armadura visible", 0,
            new ArrayList (Arrays.asList(TreasureKind.ARMOR)),
            new ArrayList (Arrays.asList()));
        Prize prize7 = new Prize (1, 1);
        unusedMonsters.add(new Monster("Necrófago", 13, badConsequence7, prize7));

        // El rey de rosado
        BadConsequence badConsequence8 = new BadConsequence ("Pierdes 5 niveles "
            + "y 3 tesoros visibles", 5, 3, 0);
        Prize prize8 = new Prize (3, 2);
        unusedMonsters.add(new Monster("El rey de rosado", 11, badConsequence8, prize8));

        // Flecher
        BadConsequence badConsequence9 = new BadConsequence ("Toses los pulmones "
            + " y pierdes dos niveles", 2, 0, 0);
        Prize prize9 = new Prize (1, 1);
        unusedMonsters.add(new Monster("Flecher", 2, badConsequence9, prize9));

        // Los hondos
        BadConsequence badConsequence10 = new BadConsequence ("Estos monstruos "
            + "resultan bastante superficiales y te aburren mortalmente. "
            + "Estas muerto", true);
        Prize prize10 = new Prize (2, 1);
        unusedMonsters.add(new Monster ("Los hondos", 8, badConsequence10, prize10));

        // Semillas Cthulthu
        BadConsequence badConsequence11 = new BadConsequence ("Pierdes 2 niveles "
            + "y 2 tesoros ocultos", 2, 0, 2);
        Prize prize11 = new Prize (2, 1);
        unusedMonsters.add(new Monster("Semillas Cthulhu", 4, badConsequence11, prize11));

        // Dameargo
        BadConsequence badConsequence12 = new BadConsequence ("Te intentas "
            + "escaquear. Pierdes una mano visible", 0,
            new ArrayList (Arrays.asList(TreasureKind.ONEHAND)),
            new ArrayList (Arrays.asList()));
        Prize prize12 = new Prize (2, 1);
        unusedMonsters.add(new Monster("Dameargo", 1, badConsequence12, prize12));

        // Pollipólipo volante
        BadConsequence badConsequence13 = new BadConsequence ("Da mucho asquito. "
            + "Pierdes 3 niveles", 3, 0, 0);
        Prize prize13 = new Prize(2, 1);
        unusedMonsters.add(new Monster("Pollipólipo volante", 3, badConsequence13, prize13));

        // Yskhtihyssg-Goth
        BadConsequence badConsequence14 = new BadConsequence ("No le hace gracia "
            + "que pronuncie mal su nombre. Estas muerto", true);
        Prize prize14 = new Prize (3, 1);
        unusedMonsters.add(new Monster("Yskhtihyssg-Goth", 14, badConsequence14, prize14));

        // Familia feliz
        BadConsequence badConsequence15 = new BadConsequence ("La familia te "
            + "atrapa. Estas muerto", true);
        Prize prize15 = new Prize (3, 1);
        unusedMonsters.add(new Monster("Familia feliz", 1, badConsequence15, prize15));

        // Roboggoth
        BadConsequence badConsequence16 = new BadConsequence ("La quinta "
            + "directiva primaria te obliga a perder 2 niveles y un tesoro "
            + "2 manos visible", 0,
            new ArrayList (Arrays.asList(TreasureKind.BOTHHANDS)),
            new ArrayList (Arrays.asList()));
        Prize prize16 = new Prize (2, 1);
        unusedMonsters.add(new Monster("Roboggoth", 8, badConsequence16, prize16));

        // El espía sordo
        BadConsequence badConsequence17 = new BadConsequence ("Te asusta en la "
            + "noche. Pierdes un casco visible", 0,
            new ArrayList (Arrays.asList(TreasureKind.HELMET)),
            new ArrayList (Arrays.asList()));
        Prize prize17 = new Prize (1, 1);
        unusedMonsters.add(new Monster("El espía sordo", 5, badConsequence17, prize17));

        // Tongue
        BadConsequence badConsequence18 = new BadConsequence ("Menudo susto te "
            + "llevas. Pierdes 2 niveles y 5 tesoros visibles", 2, 5, 0);
        Prize prize18 = new Prize (2, 1);
        unusedMonsters.add(new Monster("Tongue", 19, badConsequence18, prize18));

        // Bicéfalo
        BadConsequence badConsequence19 = new BadConsequence ("Te faltan manos "
            + "para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles"
            + "de las manos", 3,
            new ArrayList (Arrays.asList(TreasureKind.ONEHAND, TreasureKind.BOTHHANDS)),
            new ArrayList (Arrays.asList()));
        Prize prize19 = new Prize (2, 1);
        unusedMonsters.add(new Monster("Bicéfalo", 21, badConsequence19, prize19));      
    }
    
    private void shuffleTreasures(){
        Collections.shuffle(this.unusedTreasures);
    }
    
    private void shuffleMonsters(){
        Collections.shuffle(this.unusedMonsters);
    }
    
    public CardDealer getInstance(){
        return this.instance;
    }
    
    /*
    public Treasure nextTreasure(){
        
    }
    
    public Monster nextMonster(){
        
    }
    */
    
    public void giveTreasureBack(Treasure treasure){
        this.usedTreasures.add(treasure);
    }
    
    public void giveMonsterBack(Monster monster){
        this.usedMonsters.add(monster);
    }
    
    public void initCards(){
        
    }
}

File: P_2/Napakalaki_Ruby_P2/lib/card_dealer.rb
#encoding: utf-8
# MADE BY: JOSE MIGUEL HDEZ GARCIA & JAIRO ABRIL MOYA

require "singleton"

class CardDealer

    include Singleton

    attr_accesor :used_monsters
    attr_accesor :unused_monsters
    attr_accesor :used_treasures
    attr_accesor :unused_treasures

    @@unused_treasures = Array.new()
    @@used_treasures = Array.new()
    @@unused_monsters = Array.new()
    @@used_monsters = Array.new()

    def init_treasure_card_deck
        # Si mi amo
        @unused_treasures << Treasure.new("¡Si mi amo!",4, HELMET)

        # Botas de investigacion
        @unused_treasures << Treasure.new("Botas de investigación", 3,SHOES)

        # Capucha de Cthulhu
        @unused_treasures << Treasure.new("Capucha de Cthulhu", 3, HELMET)

        # A prueba de babas
        @unused_treasures << Treasure.new("A prueba de babas", 2, ARMOR)

        # Botas de lluvia acida
        @unused_treasures << Treasure.new("Botas de lluvia acida", 1, BOTHHAND)

        # Casco minero
        @unused_treasures << Treasure.new("Casco minero", 2, HELMET)

        # Ametralladora ACME
        @unused_treasures << Treasure.new("Ametralladora ACME", 4, BOTHHAND)

        # Camiseta de la ETSIIT
        @unused_treasures << Treasure.new("Camiseta de la ETSIIT", 1, ARMOR)

        # Clavo de rail ferroviario
        @unused_treasures << Treasure.new("Clavo de rail ferroviario", 3, ONEHAND)

        # Cuchillo de sushi arcano
        @unused_treasures << Treasure.new("Cuchillo de sushi arcano", 2, ONEHAND)

        # Fez alopodo
        @unused_treasures << Treasure.new("Fez alopodo", 3, HELMET)

        # Hacha prehistorica
        @unused_treasures << Treasure.new("Hacha prehistórica", 2, ONEHAND)

        # El aparato del Pr. Tesla
        @unused_treasures << Treasure.new("El aparato del Pr. Tesla", 4, ARMOR)

        # Gaita
        @unused_treasures << Treasure.new("Gaita", 4, BOTHHAND)

        # Insecticida
        @unused_treasures << Treasure.new("Insecticida", 2, ONEHAND)

        # Escopeta de 3 cañones
        @unused_treasures << Treasure.new("Escopeta de 3 cañones", 4, BOTHHAND)

        # Garabato Mistico
        @unused_treasures << Treasure.new("Garabato Mistico", 2, ONEHAND)

        # La rebeca metalica
        @unused_treasures << Treasure.new("La rebeca metálica", 2, ARMOR)

        # Mazo de los antiguos
        @unused_treasures << Treasure.new("Mazo de los antiguos", 3, ONEHAND)

        # Necroplayboycon
        @unused_treasures << Treasure.new("Necroplayboycon", 3, ONEHAND)

        # Lanzallamas
        @unused_treasures << Treasure.new("Lanzallamas", 4, BOTHHAND)

        # Necrocomicon
        @unused_treasures << Treasure.new("Necrocomicón", 1, ONEHAND)

        # Necronomicon
        @unused_treasures << Treasure.new("Necronomicón", 5, BOTHHAND)

        # Linterna a 2 manos
        @unused_treasures << Treasure.new("Linterna a 2 manos", 3, BOTHHAND)

        # Necrognomico
        @unused_treasures << Treasure.new("Necrognomicón", 2, ONEHAND)

        # Necrotelecom
        @unused_treasures << Treasure.new("Necrotelecom", 2, HELMET)

        # Porra preternatural
        @unused_treasures << Treasure.new("Porra preternatural", 2, ONEHAND)

        # Tentaculo de pega
        @unused_treasures << Treasure.new("Tentáculo de pega", 2, HELMET)

        # Zapato deja-amigos
        @unused_treasures << Treasure.new("Zapato deja-amigos", 1, SHOES)

        # Shogulador
        @unused_treasures << Treasure.new("Shogulador", 1, BOTHHAND)

        # Varita de atizamiento
        @unused_treasures << Treasure.new("Varita de atizamiento", 3, ONEHAND)
    end

    def init_monsters_card_deck

        # 3 Byakhees de bonanza
        prize = Prize.new(2,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('Pierdes tu armadura
                     visible y otra oculta', 0, [TreasureKind::ARMOR],
                     [TreasureKind::ARMOR])
        monster = Monster.new('3 Byakhees de bonanza', 8, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Tenochtitlan
        prize = Prize.new(1,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('Embobados con el
                     lindo primigenio te descartas de tu cascovisible', 0,
                     [TreasureKind::HELMET], [])
        monster = Monster.new('Tenochtitlan', 2, badconsequence, prize)
        @@unused_monsters.push(monster)

        # El sopor de Dunwich
        prize = Prize.new(1,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('El primordial
                     bostezo contagioso. Pierdes el calzado visible', 0,
                     [TreasureKind::HELMET], [])
        monster = Monster.new('El sopor de Dunwich', 2, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Demonios de Magaluf
        prize = Prize.new(4,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('Te atrapan para
                     llevarte de fiesta y te dejan caer en mitad del vuelo.
                     Descarta 1 mano visible y 1 mano oculta', 0,
                     [TreasureKind::ONEHAND], [TreasureKind::ONEHAND])
        monster = Monster.new('Demonios de Magaluf', 2, badconsequence, prize)
        @@unused_monsters.push(monster)

        # El gorron en el umbral
        prize = Prize.new(3,1)
        badconsequence = BadConsequence.newLevelNumberOfTreasures('Pierdes todos
                     tus tesoros visibles', 0, 5, 0)
        monster = Monster.new('El gorron en el umbral', 13, badconsequence, prize)
        @@unused_monsters.push(monster)

        # H.P. Munchcraft
        prize = Prize.new(2,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('Pierdes la armadura
                     visible', 0, [TreasureKind::ARMOR], [])
        monster = Monster.new('H.P. Munchcraft', 6, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Necrófago
        prize = Prize.new(1,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('Sientes bichos
                     bajo la ropa. Descarta la armadura visible', 0,
                     [TreasureKind::ARMOR], [])
        monster = Monster.new('Necrofago', 13, badconsequence, prize)
        @@unused_monsters.push(monster)

        # El rey de rosado
        prize = Prize.new(3,2)
        badconsequence = BadConsequence.newLevelNumberOfTreasures('Pierdes 5 niveles
                    y 3 tesoros visibles', 5, 3, 0)
        monster = Monster.new('El rey de rosado', 11, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Flecher
        prize = Prize.new(1,1)
        badconsequence = BadConsequence.newLevelNumberOfTreasures('Toses los pulmones
                     y pierdes 2 niveles', 2, 0, 0)
        monster = Monster.new('Flecher', 2, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Los hondos
        prize = Prize.new(2,1)
        badconsequence = BadConsequence.newDeath('Estos monstruos resultan
                      bastante superficiales y te aburren mortalmente.
                      Estas muerto')
        monster = Monster.new('Los hondos', 8, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Semillas Cthulhu
        prize = Prize.new(2,1)
        badconsequence = BadConsequence.newLevelNumberOfTreasures('Pierdes 2 niveles
                      y 2 tesoros ocultos', 2, 0, 2)
        monster = Monster.new('Semillas Cthulhu', 4, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Dameargo
        prize = Prize.new(2,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('Te intentas
                      escaquear. Pierdes una mano visible', 0,
                      [TreasureKind::ONEHAND], [])
        monster = Monster.new('Dameargo', 1, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Pollipolipo volante
        prize = Prize.new(2,1)
        badconsequence = BadConsequence.newLevelNumberOfTreasures('Da mucho asquito.
                      Pierdes 3 niveles', 3, 0, 0)
        monster = Monster.new('Pollipolipo volante', 3, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Yskhtihyssg-Goth
        prize = Prize.new(3,1)
        badconsequence = BadConsequence.newDeath('No le hace gracia que
                      pronuncien mal su nombre. Estas muerto')
        monster = Monster.new('Ysjhtihyssg-Goth', 14, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Familia feliz
        prize = Prize.new(3,1)
        badconsequence = BadConsequence.newDeath('La familia te atrapa. Estas muerto')
        monster = Monster.new('Familia feliz', 1, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Roboggoth
        prize = Prize.new(2,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('La quinta
                      directiva primaria te obliga a perder 2 niveles y un
                      tesoro 2 manos visible', 2, [], [TreasureKind::BOTHHANDS])
        monster = Monster.new('Roboggoth', 8, badconsequence, prize)
        @@unused_monsters.push(monster)

        # El espia sordo
        prize = Prize.new(1,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('Te asusta en la
                      noche. Pierdes un casco visible', 0, [TreasureKind::HELMET],
                      [])
        monster = Monster.new('El espia sordo', 5, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Tongue
        prize = Prize.new(2,1)
        badconsequence = BadConsequence.newLevelNumberOfTreasures('Menudo susto te
                      llevas. Pierdes 2 niveles y 5 tesoros visibles', 2, 5, 0)
        monster = Monster.new('Tongue', 19, badconsequence, prize)
        @@unused_monsters.push(monster)

        # Bicefalo
        prize = Prize.new(2,1)
        badconsequence = BadConsequence.newLevelSpecificTreasures('Te faltan manos
                      para tanta cabeza. Pierdes 3 niveles y tus tesoros visibles
                      de las manos', 0, [TreasureKind::BOTHHANDS, TreasureKind::ONEHAND], [])
        monster = Monster.new('Bicefalo', 21, badconsequence, prize)
        @@unused_monsters.push(monster)
    end

     #Baraja el mazo de cartas de tesoros unusedTreasures
    def shuffle_treasures
        @unused_treasures = @unused_treasures.shuffle
    end

    def shuffle_monsters
        @unused_monsters = @unused_monsters.shuffle
    end

    def next_treasure

    end

    def next_monster

    end

    def give_treasure_back(treasure)
        @used_treasures << treasure
    end

    def give_monster_back(monster)
        @used_monsters << monster
    end

    def init_cards

    end
end
